crates/ide/src/syntax_highlighting.rs

```diff
@@ -599,7 +599,10 @@ fn highlight_element(
             _ if element.parent().and_then(ast::Attr::cast).is_some() => {
                 HighlightTag::Attribute.into()
             }
-            _ => HighlightTag::Punctuation.into(),
+            k => match punctuation_modifiers(k) {
+                Some(modifier) => Highlight::new(HighlightTag::Punctuation) | modifier,
```

Comment:
Hm, I think we should have `Tag::Punctuation(Bracket | Brace ...)`, abusing modifiers for this *at this level* feels wrong.

But probably the best way to lower that to lsp in lsp_conv is indeed via modifiers: IIRC, they planned, but didn't ship inheritance for highlight tags. so we can't have `punctuation.brace`, where both things are tags. But better to double-check this.